lms/djangoapps/discussion/views.py

```diff
@@ -671,9 +691,20 @@ def render_to_fragment(self, request, course_id=None, discussion_id=None, thread
 
             fragment = Fragment(html)
             self.add_fragment_resource_urls(fragment)
-            fragment.add_javascript(inline_js)
+
+            if request.GET.get('format', 'html') == 'json':
+                fragment.add_javascript_url(staticfiles_storage.url('common/js/vendor/require.js'))
+                fragment.add_javascript_url(staticfiles_storage.url('lms/js/require-config.js.js'))
```

Comment:
@bogdanl This might be the place that causes you trouble. `require-config.js` does not have discussion JS definitions, if I recall right. You might want to introduce a separate, discussion-specific config and add it here as well.

A better approach would be to make discussions always use require.js, but that's probably outside of the scope of current task.

Reply:
Aren't the discussion JS definitions being added with these two lines? 
`inline_js = render_to_string('discussion/discussion_board_js.template', context)`
`fragment.add_javascript(inline_js)`

Reply:
@bogdanl Yes, you're right (that's the effect of delaying the task for couple of months - everyone forgets everything).

Well, since discussions requireJS config is already included, but Apros fails to use it the only remaining options are:

1. Solve the problem using nginx - and it's not as bad asit sounds, since there are already [discussion-specific rules in it](https://github.com/mckinseyacademy/mcka_apros/blob/master/docs/mcka_apros_production#L45)
2. Somehow provide an overriding require.js config in Apros or jquery-xblock